code/modules/modular_computers/file_system/programs/generic/file_browser.dm

```diff
@@ -197,4 +205,23 @@
 			if(!F || !istype(F))
 				return 1
 			var/datum/computer_file/C = F.clone(0)
-			HDD.store_file(C)
\ No newline at end of file
+			HDD.store_file(C)
+		if("PRG_encrypt")
+			. = 1
+			if(!HDD)
+				return 1
+			var/datum/computer_file/F = HDD.find_file_by_name(href_list["name"])
+			if(!F || F.undeletable)
+				return 1
+			if(F.password)
+				return
+			F.password = sanitize(input(usr, "Enter an encryption key:", "Encrypt File"))
```

Comment:
You shouldn't directly assign the result of sanitize(input(, as it could be blank. It'd be better to do this as
```
var/new_password = sanitize(input(usr, "Enter an encryption key:", "Encrypt File") as text|null))
if(!new_password)
    to_chat(usr, "<span class='warning'>File not encrypted.</span>")
    return
F.password = new_password
```